File: ruby/fizz_buzz.rb
# https://leetcode.com/problems/fizz-buzz/

def fizz_buzz(n)
  output = []

  n.times do |num|
    num += 1
    result = num.to_s
    result = 'Fizz' if num % 3 == 0
    result = 'Buzz' if num % 5 == 0
    result = 'FizzBuzz' if (num % 3 == 0) && (num % 5 == 0)
    output << result
  end

  output
end

File: python/fizz_buzz.py
# https://leetcode.com/problems/fizz-buzz/

class Solution:
    def fizzBuzz(self, n: int):
      result = []
      x = 1
      while x <= n:
        if x % 3 == 0 and x % 5 == 0:
          result.append("FizzBuzz")
        elif x % 3 == 0:
          result.append("Fizz")
        elif x % 5 == 0:
          result.append("Buzz")
        else:
          result.append(str(x))

        x += 1

      return result

result = Solution.fizzBuzz(Solution, 1)
print(result)